support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -220,19 +237,40 @@ class TestUserFromRequest(identityService: IdentityService, testUsers: TestUserS
 
 }
 
-class GetCorporateCustomer(dynamoLookup: DynamoTableAsyncForUser) {
+class CodeValidator(zuoraLookupService: ZuoraGiftLookupService, dynamoLookup: DynamoTableAsyncForUser) {
+
+  def validate(inputCode: String, isTestUser: Boolean)(implicit ec: ExecutionContext): EitherT[Future, String, ReaderType] =
+    EitherT(getValidationResult(inputCode, isTestUser).map {
+      case ValidCorporateCode(_) => Right(Corporate)
+      case ValidGiftCode(_) => Right(Gift)
+      case invalidCode => Left(validationResultToErrorMessage(invalidCode))
```

Comment:
add explicit type to make sure we get a warning if we add another Valid[....] and haven't added to the pattern match
also consider inlining this too
```suggestion
      case CodeAlreadyUsed => Left("This code has already been redeemed")
      case CodeExpired => Left("This code has expired")
      case _: InvalidCode => Left("Please check the code and try again")
```